base/strings/search.jl

```diff
@@ -266,7 +266,7 @@ function _search(s::Union{AbstractString,AbstractVector{<:Union{Int8,UInt8}}},
     idx = _searchindex(s,t,i)
     if isempty(t)
         idx:idx-1
-    elseif idx > 0
+    elseif idx > firstindex(s) - 1
```

Comment:
```suggestion
    elseif idx >= firstindex(s)
```

Reply:
(Since `_nthbyte` also computes `firstindex(s) - 1`, leaving it as `idx > firstindex(s) - 1` preserves the common subexpression.  Not that it should matter much, since in most cases `firstindex(s)` should be a constant and the compiler will turn both variants into exactly the same code.)